server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/node/InMemoryTreeNode.java

```diff
@@ -63,6 +62,11 @@ public class InMemoryTreeNode {
      */
     public Map<Long, InMemoryTreeNode> children;
 
+    /**
+     * A map containing the child star node of this star-tree node.
+     */
+    public InMemoryTreeNode childStarNode;
```

Comment:
Make this private and expose a getter on it. Lets do it for all the attributes of this class. 
I missed this comment in the earlier PRs.

Certain things like how we are maintaining children, or star node child should not be leaked to any component outside of this class.

Reply:
This change is causing a bigger blast radius in BaseStarTreeBuilder, StarTreeDataWriter, and many other tests classes. Since this is not a much needed design change (I believe), I would recommend taking it in a separate PR as this PR is 3200 lines long.

Reply:
Sure, please create an issue for this and we can assign it to you. We need to handle such changes properly.

Reply:
> Sure, please create an issue for this and we can assign it to you. We need to handle such changes properly.

I disagree. We need to ensure we are raising the bar on coding practices alongside correctness, and hence the ask. 

> I would recommend taking it in a separate PR as this PR is 3200 lines long.



Sure, please create an issue for this and we can assign it to you. We need to handle such changes properly.

Reply:
> I disagree. We need to ensure we are raising the bar on coding practices alongside correctness, and hence the ask.

Since you feel strongly about the ask, I would be taking the change in this PR itself. I am totally against lowering the bar in any of the merge in OpenSearch :)